core/src/main/java/hudson/PluginManager.java

```diff
@@ -423,6 +434,160 @@ public void run(Reactor reactor) throws Exception {
         }});
     }
 
+    protected @Nonnull Set<String> loadPluginsFromWar(@Nonnull String fromPath) {
+        return loadPluginsFromWar(fromPath, null);
+    }
+
+    protected @Nonnull Set<String> loadPluginsFromWar(@Nonnull String fromPath, @CheckForNull FilenameFilter filter) {
+        Set<String> names = new HashSet();
+
+        ServletContext context = Jenkins.getInstance().servletContext;
+        Set<String> plugins = Util.fixNull((Set<String>) context.getResourcePaths(fromPath));
+        Set<URL> copiedPlugins = new HashSet<>();
+        Set<URL> dependencies = new HashSet<>();
```

Comment:
FindBugs will indeed warn about it. The problem is very real if the URLs are, for example, `http` protocol, because `URL.equals` does a DNS comparison on hostnames, so just adding to a collection could block on network I/O! Dumb, right? (`URI` uses a more superficial comparison and so is safe.) In this case I think the URLs are always local so it is not an issue.